### PR TITLE
finding 46 add erc165

### PR DESCRIPTION
**Pull request by livingrockrises**
need to check which all interfaces we need to add to be compliant with the ERC7579.
it's not there in reference implementation so I would probably open a PR there too and get feedback.

### CODE REVIEW

File: contracts/Nexus.sol [hunk 1]
```diff
@@ -266,6 +269,16 @@ contract Nexus is INexus, BaseAccount, ExecutionHelper, ModuleManager, UUPSUpgra
         else return false;
     }
 
+    /**
+     * @dev See {IERC165-supportsInterface}.
+     */
+    function supportsInterface(bytes4 interfaceId) public view virtual override returns (bool) {
+        return
+            interfaceId == type(IERC165).interfaceId ||
```
Comment (Aboudjem) [inline contracts/Nexus.sol:277]: No need to check for ERC165 already implements it


```
    function supportsInterface(bytes4 interfaceId) public view virtual returns (bool) {
        return interfaceId == type(IERC165).interfaceId;
    }
```

**Review by Aboudjem — CHANGES_REQUESTED** [review]
tests are missing + would i make sense to add IERC4337Account and ERC7484 ?

File: contracts/Nexus.sol (hunk 1 above)
Comment (livingrockrises) [reply contracts/Nexus.sol:277]: yeah but I am overriding and want to add more types.

**Comment by livingrockrises** [comment]
> tests are missing + would i make sense to add IERC4337Account and ERC7484 ?

yeah. we should / will make a list of missing tests

interfaces I need to confirm with Rhinestone. 

why did you even suggest ERC7484? 
issue is specific to ERC7579 needs of supported interfaces. 
https://cantina.xyz/code/d1d4b139-9705-4367-9468-297b7078674e/findings/46

feel it's merely IERC7579Account , (maybe IERC54337Account)

**Comment by livingrockrises** [comment]
> > > tests are missing + would i make sense to add IERC4337Account and ERC7484 ?
> > 
> > 
> > yeah. we should / will make a list of missing tests
> > interfaces I need to confirm with Rhinestone.
> > why did you even suggest ERC7484? issue is specific to ERC7579 needs of supported interfaces. https://cantina.xyz/code/d1d4b139-9705-4367-9468-297b7078674e/findings/46
> > feel it's merely IERC7579Account , (maybe IERC54337Account)
> 
> for the factories I meant

no this is ERC165 for the account what interfaces must be supported.

**Comment by livingrockrises** [comment]
I am gonna potentially close this if we make do using fallback handler.

**Comment by livingrockrises** [comment]
closing this as we would do this via fallback handler as discussed with zeroknots.
need votes
@filmakarov @VGabriel45 @Aboudjem

**Comment by livingrockrises** [comment]
closing this as of now..

**Pull request closed without merging**